samza-test/src/main/python/tests/standalone_failure_tests.py

```diff
@@ -0,0 +1,306 @@
+
+        leader_processor_id = zk_client.get_leader_processor_id()
+        processors.pop(leader_processor_id).kill()
+
+        for processor_id in processors.keys():
+            processors.pop(processor_id).kill()
+            break
+
+        event = threading.Event()
+        zk_client.watch_job_model(job_model_watch(event = event, expected_processors=processors.keys()))
+
+        event.wait(GROUP_COORDINATION_TIMEOUT * 2)
+
+        ## Verifications after killing the processors.
+        job_model = zk_client.get_latest_job_model()
+        for processor_id in processors.keys():
+            assert processor_id in job_model['containers'], 'Processor id: {0} does not exist in JobModel: {1}.'.format(processor_id, job_model)
+        assert leader_processor_id not in job_model['containers'], 'Leader processor id: {0} exists in JobModel: {1}.'.format(leader_processor_id, job_model)
+    except:
+        ## Explicitly logging exception, since zopkio doesn't log complete stacktrace.
+        logger.error(traceback.format_exc(sys.exc_info()))
+        raise
+    finally:
+        __tear_down_processors(processors)
+        __purge_zk_data()
+        __teardown_zk_client()
+
+def test_pause_resume_leader():
+    """
+    Launches three processors. Pauses the leader processor. Wait till group coordination timeout and verifies that the
+    JobModel doesn't contain leader processor. Resumes the leader processor and waits till group coordination timeout,
+    verifies that new JobModel contains the previously paused leader processor.
+    """
+    processors = {}
+    try:
+        __setup_zk_client()
+        __pump_messages_into_input_topic()
+        processors = __setup_processors()
+
+        event = threading.Event()
+        zk_client.watch_job_model(job_model_watch(event = event, expected_processors=processors.keys()))
+
+        event.wait(GROUP_COORDINATION_TIMEOUT * 2)
+
+        ## First JobModel generation.
+        job_model = zk_client.get_latest_job_model()
+        for processor_id, deployer in processors.iteritems():
+            assert processor_id in job_model['containers'], 'Processor id: {0} does not exist in JobModel: {1}.'.format(processor_id, job_model)
+
+        leader_processor_id = zk_client.get_leader_processor_id()
+        leader = processors.pop(leader_processor_id)
+
+        logger.info("Pausing the leader processor: {0}.".format(leader_processor_id))
+        leader.pause()
+
+        event = threading.Event()
+        zk_client.watch_job_model(job_model_watch(event = event, expected_processors=processors.keys()))
+
+        event.wait(GROUP_COORDINATION_TIMEOUT * 2)
+
+        ## Verifications after leader was suspended.
+        job_model = zk_client.get_latest_job_model()
+        for processor_id, deployer in processors.iteritems():
+            assert processor_id in job_model['containers'], 'Processor id: {0} does not exist in containerModel: {1}.'.format(processor_id, job_model['containers'])
```

Comment:
this asserts that all processors except the leader are on the group. should we also assert that the leader is _not_ in the group?

Reply:
Done.